src/main/kotlin/org/opensearch/replication/util/ValidationUtil.kt

```diff
@@ -138,4 +141,17 @@ object ValidationUtil {
             throw validationException
         }
     }
+
+    /**
+     * Throw exception if leader index is knn a knn is not installed
+     */
+    fun checkKNNEligibility(leaderSettings: Settings, clusterService: ClusterService, leaderIndex: String) {
+        if(leaderSettings.getAsBoolean(KNN_INDEX_SETTING, false)) {
```

Comment:
During resume, we are only depend on follower index settings irrespective of leader index settings as the index is already replicated. 

and can we check the index replication task flow as well?

Reply:
Test the resume flow, the resume turns the replicating index red